### PR TITLE
Update upcloud_server.html.markdown

### PR DESCRIPTION
**Pull request by Darep**
This PR:

- adds "Attribute References"-section
- moves required attributes to the top
- updates the example to match current preferred way of working
- small corrections here and there

Related issue: #89

### CODE REVIEW

File: website/docs/r/upcloud_server.html.markdown [hunk 1]
```diff
@@ -56,34 +52,22 @@ The following arguments are supported:
...
 * `mem` - (Optional) The size of memory for the server
 * `template` - (Optional) The template to use during creation
 * `user_data` - (Optional) Defines URL for a server setup script, or the script body itself
 * `plan` - (Optional) The pricing plan used for the server
-* `storage_devices` - (Required) A list of storage devices associated with the server
+* `storage_devices` - (Optional) A list of storage devices associated with the server
 * `login` - (Optional) Configure access credentials to the server
-* `network_interface` - (Required) One or more blocks describing the network interfaces of the server. Any changes to these blocks will force the creation of a new server resource.
 
 The `storage_devices` block supports:
```
Comment (tohalla) [inline website/docs/r/upcloud_server.html.markdown:66]: seems that the template block is missing similar description

**Review by tohalla — CHANGES_REQUESTED** [review]
Good stuff, good to go once tamplate block described

File: website/docs/r/upcloud_server.html.markdown (hunk 1 above)
Comment (Darep) [reply website/docs/r/upcloud_server.html.markdown:66]: Good catch! Added now ☺️

**Comment by Darep** [comment]
Merging!

**Pull request merged**